src/device-state.ts

```diff
@@ -854,7 +854,7 @@ export function triggerApplyTarget({
 			// If a delay has been set it's because we need to hold off before applying again,
 			// so we need to respect the maximum delay that has
 			// been passed
-			if (!scheduledApply.delay) {
+			if (scheduledApply.delay === undefined || isNaN(scheduledApply.delay)) {
```

Comment:
This could be:
```suggestion
			if (isNaN(scheduledApply.delay)) {
```
and we should also show what the value itself is in the error message

Reply:
I agree this is the desired logic but the **delay** value can be undefined as well and isNaN expects a number so we must check first so the compiler can infer it is a number when passing it to isNaN. The typing seems kinda off doesn't it ? You'd expect isNaN to accept type of any.

Also I was confident the value being logged will be undefined and isNaN was used just to be more strict. I don't see how it will ever be anything but a number or undefined however I will never say no to adding more insight on whats going on with the device :p